src/org/rust/cargo/project/CargoProjectResolver.kt

```diff
@@ -0,0 +1,155 @@
+package org.rust.cargo.project
+
+import com.google.gson.Gson
+import com.intellij.execution.ExecutionException
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.CapturingProcessHandler
+import com.intellij.execution.process.ProcessAdapter
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessOutputTypes
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.externalSystem.model.DataNode
+import com.intellij.openapi.externalSystem.model.ExternalSystemException
+import com.intellij.openapi.externalSystem.model.ProjectKeys
+import com.intellij.openapi.externalSystem.model.project.*
+import com.intellij.openapi.externalSystem.model.task.ExternalSystemTaskId
+import com.intellij.openapi.externalSystem.model.task.ExternalSystemTaskNotificationEvent
+import com.intellij.openapi.externalSystem.model.task.ExternalSystemTaskNotificationListener
+import com.intellij.openapi.externalSystem.service.project.ExternalSystemProjectResolver
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.vfs.VfsUtil
+import org.apache.commons.lang.StringUtils
+import org.rust.cargo.Cargo
+import org.rust.cargo.project.model.CargoProjectInfo
+import org.rust.cargo.project.settings.CargoExecutionSettings
+import org.rust.cargo.service.CargoInstallationManager
+import org.rust.lang.module.RustModuleType
+import java.io.File
+import java.util.*
+
+class CargoProjectResolver : ExternalSystemProjectResolver<CargoExecutionSettings> {
+
+    @Throws(ExternalSystemException::class, IllegalArgumentException::class, IllegalStateException::class)
+    override fun resolveProjectInfo(id: ExternalSystemTaskId,
+                                    projectPath: String,
+                                    isPreviewMode: Boolean,
+                                    settings: CargoExecutionSettings?,
+                                    listener: ExternalSystemTaskNotificationListener): DataNode<ProjectData>? {
+
+        listener.onStatusChange(ExternalSystemTaskNotificationEvent(id, "Resolving dependencies..."))
+
+        val data: CargoProjectInfo
+        try {
+            val cargoInstallationManager = ServiceManager.getService(CargoInstallationManager::class.java)
+
+            val processOut =
+                cargoInstallationManager.runExecutableWith(
+                    settings!!.cargoExecutable,
+                    arrayListOf(
+                        "metadata",
+                        "--output-format",  "json",
+                        "--manifest-path",  File(projectPath, Cargo.BUILD_FILE).absolutePath,
+                        "--features",       StringUtils.join(settings.features, ",")
+                    ),
+                    object : ProcessAdapter() {
+                        override fun onTextAvailable(event: ProcessEvent, outputType: Key<Any>) {
+                            val text = event.text.trim { it <= ' ' }
```

Comment:
`it <= ' ' ` look suspicious. Am I correct that this removes all characters with ascii code bellow 32?